projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -2124,71 +1742,187 @@ export class IgxTimePickerComponent implements
         });
     }
 
-    private _onDropDownClosed(): void {
-        const oldValue = this.value;
-        const newVal = this.convertMinMaxValue(this.displayValue);
+    private parseHours(value: Date, format: string): string {
+        let hour = value.getHours();
+        let formattedHour;
 
-        if (this.displayValue === this.parseMask(false)) {
+        if (format.indexOf('h') !== -1) {
+            hour = hour % 12;
+            hour = hour ? hour : 12; // the hour '0' should be '12'
+            formattedHour = hour < 10 && format.indexOf('hh') !== -1 ? '0' + hour : `${hour}`;
+        } else {
+            formattedHour = hour < 10 && format.indexOf('HH') !== -1 ? '0' + hour : `${hour}`;
+        }
+
+        return formattedHour;
+    }
+
+    private parseMinutes(value: Date, format: string): string {
+        const minutes = value.getMinutes();
+        const formattedMinutes = minutes < 10 && format.indexOf('mm') !== -1 ? '0' + minutes : `${minutes}`;
+        return formattedMinutes;
+    }
+
+    private parseSeconds(value: Date, format: string): string {
+        const seconds = value.getSeconds();
+        const formattedSeconds = seconds < 10 && format.indexOf('ss') !== -1 ? '0' + seconds : `${seconds}`;
+        return formattedSeconds;
+    }
+
+    private parseAMPM(value: Date, format: string): string {
+        let hour = value.getHours();
+        const ampm = hour >= 12 ? 'PM' : 'AM';
+        return ampm;
+    }
+
+    private handleContainerClosed(): void {
+        const newValue = this.dateTimeEditor.value;
+        const minDate: Date = this.parseToDate(this.minValue);
+        const maxDate: Date = this.parseToDate(this.maxValue);
+
+        if (this.value.getTime() === newValue.getTime()) {
```

Comment:
Same as all the rest, if `this.value` is `string`, calling `getTime` will throw. So, instead, we can use an internal `Date` value.